### PR TITLE
path/internal: new package

### PR DESCRIPTION
**Pull request by kortschak**
Move tilegraph=>grid to path/internal and clean up.

~~NOT FOR SUBMISSION~~

This is the first stage of writing dynamic graph search tests. The tests in this package need to be cleaned up and extended a little before submission.

@Jragonmiris @vladimir-ch PTAL

### CODE REVIEW

File: path/internal/grid.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+// in the grid, (NaN, NaN) is returned.
+func (g *Grid) XY(n graph.Node) (x, y float64) {
+	if !g.Has(n) {
+		return math.NaN(), math.NaN()
+	}
+	r, c := g.RowCol(n.ID())
+	return float64(c), float64(r)
+}
+
+// NodeAt returns the at (r, c). The returned node may be open or closed.
```
Comment (vladimir-ch) [inline path/internal/grid.go]: the node at

**Comment by vladimir-ch** [comment]
LGTM

**Comment by kortschak** [comment]
Cleaned/added tests. PTAL

**Comment by vladimir-ch** [comment]
LGTM

**Pull request merged**